indico/modules/events/contributions/controllers/display.py

```diff
@@ -106,6 +105,37 @@ def _process(self):
                                                contribution=self.contrib, event=self.event_new, **ical_params)
 
 
+class RHAuthorList(RHDisplayProtectionBase):
+
+    MENU_ENTRY_NAME = 'author_index'
+
+    def _process(self):
+        authors = (EventPerson.find(ContributionPersonLink.author_type != AuthorType.none,
```

Comment:
```python
authors = (e.persons
           .filter(EventPerson.contribution_links.any(db.and_(ContributionPersonLink.author_type != AuthorType.none,
                                                              ContributionPersonLink.contribution.has(Contribution.is_deleted))))
           .options(subqueryload('contribution_links').joinedload('contribution'))
           .order_by(db.func.lower(EventPerson.last_name))
           .all())
```

And since it's almost the same for speakers I would make this a `_get_persons()` function which takes an argument for the person link filter (`ContributionPersonLink.author_type != AuthorType.none` or `ContributionPersonLink.is_speaker`)